test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -452,5 +457,65 @@ TEST(QuicPlatformTest, QuicTestOutput) {
                       QuicRecordTestOutput("quic_test_output.3", "output 3 content\n"));
 }
 
+class FileUtilsTest : public testing::Test {
+public:
+  FileUtilsTest() : dir_path_(Envoy::TestEnvironment::temporaryPath("envoy_test")) {
+    files_to_remove_.push(dir_path_);
+  }
+
+protected:
+  void SetUp() override { Envoy::TestUtility::createDirectory(dir_path_); }
+
+  void TearDown() override {
+    while (!files_to_remove_.empty()) {
+      const std::string& f = files_to_remove_.top();
+      Envoy::TestEnvironment::removePath(f);
+      files_to_remove_.pop();
+    }
+  }
+
+  void addSubDirs(std::list<std::string> sub_dirs) {
+    for (const std::string& dir_name : sub_dirs) {
+      const std::string full_path = dir_path_ + "/" + dir_name;
+      Envoy::TestUtility::createDirectory(full_path);
+      files_to_remove_.push(full_path);
+    }
+  }
+
+  void addFiles(std::list<std::string> files) {
+    for (const std::string& file_name : files) {
+      const std::string full_path = dir_path_ + "/" + file_name;
+      { const std::ofstream file(full_path); }
+      files_to_remove_.push(full_path);
+    }
+  }
+
+  const std::string dir_path_;
+  std::stack<std::string> files_to_remove_;
+};
+
+TEST_F(FileUtilsTest, ReadDirContents) {
+  addSubDirs({"sub_dir1", "sub_dir2", "sub_dir1/sub_dir1_1"});
+  addFiles({"file", "sub_dir1/sub_file1", "sub_dir1/sub_dir1_1/sub_file1_1", "sub_dir2/sub_file2"});
+
+  const std::unordered_set<std::string> expected = {
+      dir_path_ + "/file", dir_path_ + "/sub_dir1/sub_file1",
+      dir_path_ + "/sub_dir1/sub_dir1_1/sub_file1_1", dir_path_ + "/sub_dir2/sub_file2"};
+  std::vector<std::string> files = ReadFileContents(dir_path_);
+  EXPECT_EQ(expected.size(), files.size());
```

Comment:
Line 505 to 508: You can use EXPECT_THAT(files, UnorderedElementsAre(expected))

Reply:
good point!